File: LeetCode/0693. Binary Number with Alternating Bits/Solution.java
/**
 * 60 / 60 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 34.9 MB 
 */
class Solution {
    public boolean hasAlternatingBits(int n) {
        int x = n ^ (n >> 1);
        return (x & (x + 1)) == 0;
    }
}

File: LeetCode/0693. Binary Number with Alternating Bits/solution.rs
/**
 * 204 / 204 test cases passed.
 * Runtime: 0 ms
 * Memory Usage: 1.9 MB 
 */
impl Solution {
    pub fn has_alternating_bits(n: i32) -> bool {
        let x = n ^ (n >> 1);       
        x & (x + 1) == 0
    }
}